Remove Mandatory signup screen + Introduced local storage when account is not created

shaavan opened the pull request:
Resolves #331

- Removes the mandatory signup screen.
- Introduce an option for loading/saving progress locally when the user has not created the account.

shaavan commented:
The PR is correctly building and running on my system, but the Vercel deployment is failing for some reason.

@afterburn, would you please take a look into this? Thanks!

benalleng commented:
Tested on my local machine, something to look into is how we can initialize from the localprogress once logging in, at the moment it seems to reset user progress back to the beginning after logging in

afterburn commented:
Hey @shaavan thanks for working on this issue. I unfortunately dont have access to our Vercel team lol so I reached out to @adamjonas. Will get back to you asap.

shaavan commented:
@benalleng 

> At the moment it seems to reset user progress back to the beginning after logging in

Yes! This is something I thought about.
I wanted to get some initial reviews on the approach before implementing the account progress and local progress interaction.

shaavan commented:
@afterburn

> I unfortunately dont have access to our Vercel team lol so I reached out to @adamjonas. Will get back to you asap.

All right! Thanks, Kevin!

afterburn commented:
hey @shaavan, if you push another commit the vercel build should redeploy

shaavan commented:
> if you push another commit the vercel build should redeploy. I think maybe the issue was that there were too many open PR's.

Sure, let me give it a try!

shaavan commented:
Repushed the commit to restart Vercel Build.

Edit: The build is still failing 🥲

@adamjonas
A gentle ping. Please let us know the reason for the build failure in Vercel. Thanks!

shaavan commented:
Updated from b5fc8168bdfee6e94912c27e466f2334be3e55df to cabb691ad50d9cd72fa592891642cd60e490f804([pr399.03](https://github.com/shaavan/saving-satoshi/commits/pr399.03) -> [pr399.04](https://github.com/shaavan/saving-satoshi/commits/pr399.04), [diff](https://github.com/shaavan/saving-satoshi/compare/pr399.03..pr399.04))

Changes:

1. Rebased over master.
2. Fixed build issue related to:
    1. non-null assertion (`fn(progress)` -> `fn(progress!)`)
    2. bad import of Google fonts (`next/font/google` -> `@next/font/google`)

shaavan commented:
Changes:

Removed `node_modules` and reinstalled them to download the latest package versions.

shaavan commented:
Cleaned and reinstalled dependencies again after running the following instructions:

```
rm -rf node_modules
rm -rf .next
rm -rf yarn.lock

yarn install

yarn build
```

benalleng commented:
> 2\. bad import of Google fonts (`next/font/google` -> `@next/font/google`)

@shaavan I found that after properly upgrading my next version this error is actually not an error, reset your text editor and see if it still represents this as an error

shaavan commented:
@benalleng 

> I found that after properly upgrading my next version this error is actually not an error

I have noticed this, and with the latest PR update (1a33a6eeb878c68942e3c66ec471df634bbb598d), I reverted this change to work with the latest _Next_ version.

afterburn commented on api/local/setProgress.ts:
```diff
@@ -0,0 +1,8 @@
+export default async function setProgress(progress: string) {
+  try {
+    localStorage.setItem('SavingSatoshiProgress', JSON.stringify(progress))
```
Not sure why you are using `JSON.stringify` here, this will result in extra `"` characters around the lessonId. This in turn forces you to use `.replace(/['"]/g, '')` [here](https://github.com/saving-satoshi/saving-satoshi/pull/399/files#diff-1a2ca889da37ca6ad47d64eaae05a834b5886e6a8bba5b1c2ea555045c97dccdR7).

shaavan replied:
Not using Json.stringify() causes the following error.

<img width="1503" alt="image" src="https://user-images.githubusercontent.com/85434418/234794581-64c330a3-c85f-4b16-80c1-d40659ecc9a3.png">

That's why using `JSON.stringify()` while saving and using `.replace(/['"]/g, '')` while reading is the cleanest solution in our case.

afterburn replied:
Hey @shaavan I checked out your branch locally, made the changes I suggested and was unable to reproduce your issue. 

The destructuring error happens if you try to do something like this:
```ts
const { path } = undefined
```
Which I dont see you doing anywhere in your code, perhaps minification has changed the names of variables but I'm wondering why you're seeing minified code in your browser window. Can you provide some steps to reproduce?

shaavan replied:
> Can you provide some steps to reproduce?

1. I made the changes you suggested. (Same as bd82a76b651ee822a133fb0b90dd916a2fcec929)
2. Compiled and ran the branch (`npm run build` `npm run start`)
3. I navigated to [Intro-1 page](http://localhost:3000/chapters/chapter-1/intro-1) and noticed this error.

afterburn reviewed (dismissed):
Hey @shaavan, requested a small change after which I will merge your PR. I checked it out locally and couldn't find any issues so we'll ignore the failing Vercel build.

shaavan commented:
> Pushed the changes I suggested, build is now also working.

Finally, successful build!! Yay! 🥳

ecurrencyhodler commented:
tACK. Great work.
